Update cfutil.py

leilabbb opened the pull request:
related to issue#950 [https://github.com/ioos/compliance-checker/issues/950]

updated the function get_grid_mapping_variables to parse the attribute variables when it contains more than one string.

benjwadams committed:
Fix netCDF4 createVariable invocation

benjwadams committed:
WIP check root group attr names, not running tests

leilabbb committed:
Update cfutil.py

related to issue#950 [https://github.com/ioos/compliance-checker/issues/950]

updated the function get_grid_mapping_variables to parse the attribute variables when it contains more than one string.

leilabbb committed:
Update test_cf.py

related to issue # 950 [https://github.com/ioos/compliance-checker/issues/950]

Mocked a new .nc file to test the grid_mapping requirements and recommendations

leilabbb committed:
Update cfutil.py

fixed the for loop of the get_grid_mapping_variables function

leilabbb committed:
Update test_cf.py

leilabbb committed:
modification to 5.6 Grid Mappings and Projections #950

in cf_1_7.py
- added "geographic_crs_name" to the function def _check_gmattr_existence_condition_ell_pmerid_hdatum(self, var):

- Modified function def check_grid_mapping(self, ds):

in cfutil.py
- modified the function def grid_mapping_variables(ds)

- added the function def get_grid_mapping_variables_coordinates(ds):

in test_cf.py
- modified the function def test_check_grid_mapping(self):

ocefpaf committed:
add dependabot.yml

ocefpaf committed:
rename and update PyPI GHA

ocefpaf committed:
use micromamba instead of conda GHA

ocefpaf committed:
update

ocefpaf committed:
run pre-commit

ocefpaf committed:
cf_units is not available on py311 yet

benjwadams committed:
Merge pull request #980 from ocefpaf/use_micromamba_v2

Use micromamba v2

leilabbb committed:
Update test_cf.py

removed commented function

leilabbb committed:
Update cf_1_7.py

removed "print( )" to suppress output, it is needed only for debugging. 

removed the IF conditional statement and used the tests' results from  _check_gmattr_existence_condition_ell_pmerid_hdatum and _check_gmattr_existence_condition_geoid_name_geoptl_datum_name

benjwadams commented on compliance_checker/cf/cf_1_7.py:
```diff
@@ -724,17 +728,107 @@ def _evaluate_towgs84(self, val):
+                "is not a string".format(item.grid_mapping))
+                test_ctx.out_of += 1
+                if ' ' in item.grid_mapping:  
+                    for x in item.grid_mapping.replace(":", '').split():      
```
Is this segment reachable?  Consider using something like regex here.

dependabot[bot] committed:
Bump pypa/gh-action-pypi-publish from 1.8.3 to 1.8.4

Bumps [pypa/gh-action-pypi-publish](https://github.com/pypa/gh-action-pypi-publish) from 1.8.3 to 1.8.4.
- [Release notes](https://github.com/pypa/gh-action-pypi-publish/releases)
- [Commits](https://github.com/pypa/gh-action-pypi-publish/compare/v1.8.3...v1.8.4)

---
updated-dependencies:
- dependency-name: pypa/gh-action-pypi-publish
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

ocefpaf committed:
Merge pull request #1000 from ioos/dependabot/github_actions/pypa/gh-action-pypi-publish-1.8.4

Bump pypa/gh-action-pypi-publish from 1.8.3 to 1.8.4

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/psf/black: 23.1.0 → 23.3.0](https://github.com/psf/black/compare/23.1.0...23.3.0)
- [github.com/codespell-project/codespell: v2.2.2 → v2.2.4](https://github.com/codespell-project/codespell/compare/v2.2.2...v2.2.4)

dependabot[bot] committed:
Bump pypa/gh-action-pypi-publish from 1.8.4 to 1.8.5

Bumps [pypa/gh-action-pypi-publish](https://github.com/pypa/gh-action-pypi-publish) from 1.8.4 to 1.8.5.
- [Release notes](https://github.com/pypa/gh-action-pypi-publish/releases)
- [Commits](https://github.com/pypa/gh-action-pypi-publish/compare/v1.8.4...v1.8.5)

---
updated-dependencies:
- dependency-name: pypa/gh-action-pypi-publish
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

ocefpaf committed:
Merge pull request #1002 from ioos/dependabot/github_actions/pypa/gh-action-pypi-publish-1.8.5

Bump pypa/gh-action-pypi-publish from 1.8.4 to 1.8.5

ocefpaf committed:
Merge pull request #1001 from ioos/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

benjwadams committed:
[CF 1.9] Ensure time coordinate variables have a calendar attribute specified

benjwadams committed:
Only fetch time coordinate variables for calendar check

benjwadams committed:
Add tests for remaining code paths in time coord var calendar check

benjwadams committed:
Add CF conformance annotations to calendar check and test

benjwadams committed:
Exclude miscellaneous files from being checked by codespell

Excludes seanames.csv and a testing configuration file from codespell
pre-commit checking.

benjwadams committed:
Fixes dead ERDDAP link in integration tests

benjwadams committed:
Merge pull request #1004 from benjwadams/remove_misc_files_codespell

Exclude miscellaneous files from being checked by codespell

benjwadams committed:
Merge pull request #1005 from benjwadams/fix_integration_test_dead_links

Fixes dead ERDDAP link in integration tests

ocefpaf committed:
add GHA for building docs

ocefpaf committed:
stale

ocefpaf committed:
add docs

ocefpaf committed:
fix links

ocefpaf committed:
ignore docs build files

ocefpaf committed:
docs deps

ocefpaf committed:
fix links

ocefpaf committed:
fix error for numpy 1.24.2

ocefpaf committed:
update and add more words to skip

benjwadams committed:
Merge pull request #1009 from ocefpaf/fix_numpy_error

Fix numpy deprecations that are errors in 1.24.2

ocefpaf committed:
move to pyproject.toml

benjwadams committed:
Merge branch 'develop'

ocefpaf committed:
fix minor inconsistencies in GHA

benjwadams committed:
Implement majority of conformance for CF section 4.4

Adds implementations and tests for the following sections:

Required:
- The reference date/time in time units is not allowed to contain
  seconds equal to or greater than 60.

Recommendations:
- The use of time coordinates in year 0 and reference date/times in year
  0 to indicate climatological time is deprecated.
- Units of year and month and any equivalent units should be used with
  caution.

benjwadams committed:
Remove unused helper functions for axes

benjwadams committed:
Remove more unused classes in cf/util.py

benjwadams committed:
Merge pull request #1016 from benjwadams/remove_unused

Remove unused helper functions for axes

benjwadams committed:
Merge pull request #1013 from ocefpaf/fix_minor_inconsistencies

fix minor inconsistencies in GHA

benjwadams committed:
Merge pull request #1003 from benjwadams/add_time_coordinate_calendar_check

[CF 1.9] Ensure time coordinate variables have a calendar attribute s…

benjwadams committed:
Merge branch 'master' into develop

benjwadams committed:
Fix time conformance tests for CF 1.6

- Uses cftime for checking recommendation to not use "months/years since" in time interval
- Simplifies year zero climatology check
- Removes seconds >= 60 check from CF 1.6, is implemented in CF 1.9 and
  next commit will implement.

benjwadams committed:
valid_caldendars -> standard calendars, small clarifying comment

benjwadams committed:
Fix year 1582 crossover calculation

benjwadams committed:
Fix calendar subsection, add check for no calendar attr with time coord variable

benjwadams committed:
Merge branch 'develop' into add_date_time_conformance

benjwadams committed:
Handle 1582 crossover issues when invalid date provided

benjwadams committed:
[CF 1.6] Only calendar check time coordinate variables

benjwadams committed:
Handle calendar check for time coordinate variables

Handles checking for calendar in time coordinate variable.

- In CF < 1.9, calendar is not required, so a dummy return is used.
- In CF >= 1.9, calendar is required for time coordinate variables and
  is checked.
- Add CF >= 1.9 check for time units >= 60 seconds.

benjwadams committed:
Remove superfluous test helper class attributes

benjwadams committed:
Only check calendar presence for time coord vars in CF 1.9 onwards

benjwadams committed:
Fix/add year zero check to section 4.4.1/calendar

benjwadams committed:
Merge pull request #1014 from benjwadams/add_date_time_conformance

Implement majority of conformance for CF section 4.4

benjwadams committed:
Check strict monotonicity of coordinate variables

benjwadams committed:
Merge pull request #1017 from benjwadams/coordinate_variable_monotonicity

Check strict monotonicity of coordinate variables

leilabbb committed:
Update cf_1_7.py

using  regex I have replaced an inner loop with:
grid_mapping_variable_list.append(regex.findall(r"(\w+)",item.grid_mapping))

benjwadams commented on compliance_checker/cf/cf_1_7.py:
```diff
@@ -741,13 +741,11 @@ def check_grid_mapping(self, ds):
+                
+                # [2/9] Note that in its simplest form the attribute comprises just 
+                # a grid_mapping_name as a single word.
+                grid_mapping_variable_list.append(regex.findall(r"(\w+)",item.grid_mapping))
```
This still falls through due to `if` clause above.  If it's not a string, it will fail out of the function when attempting to match against a non-string object.  If it is a string, the `regex` section will never be reached.  Perhaps put in an `else` clause or use a `continue` statement. If you're trying to *just* get the `grid_mapping` variable, you might try `regex.match(r"\w+")` since it implicitly anchors to start of string and won't fetch anything at the comma or beyond, so will skip coordinate variables in the latter form.

leilabbb replied:
added the else clause .... I missed it ! it was in the logic before I made the change

dependabot[bot] committed:
Bump pypa/gh-action-pypi-publish from 1.8.5 to 1.8.6

Bumps [pypa/gh-action-pypi-publish](https://github.com/pypa/gh-action-pypi-publish) from 1.8.5 to 1.8.6.
- [Release notes](https://github.com/pypa/gh-action-pypi-publish/releases)
- [Commits](https://github.com/pypa/gh-action-pypi-publish/compare/v1.8.5...v1.8.6)

---
updated-dependencies:
- dependency-name: pypa/gh-action-pypi-publish
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

ocefpaf committed:
Merge pull request #1018 from ioos/dependabot/github_actions/pypa/gh-action-pypi-publish-1.8.6

Bump pypa/gh-action-pypi-publish from 1.8.5 to 1.8.6

benjwadams committed:
Merge branch 'develop' into add_tests_groups

benjwadams committed:
Fix tests again non-root group invalid attributes

benjwadams committed:
Merge pull request #921 from benjwadams/add_tests_groups

WIP check root group attr names, not running tests

benjwadams committed:
Refactor cell_measures, properly handling quantity squared/cubed

benjwadams committed:
Fix test example with undefined units for cell measures variable

leilabbb committed:
Update cf_1_7.py

added the "else" clause to the "if not" statement on line 740

benjwadams committed:
Change cell_methods error message, improve test coverage

benjwadams committed:
Merge pull request #1019 from benjwadams/fix_cell_measure_quantity_squared_cubed

Refactor cell_measures, properly handling quantity squared/cubed

ocefpaf committed:
fix names, update GHAs, remove cruft

ocefpaf committed:
update and add extra checks

ocefpaf committed:
skip yaml

ocefpaf committed:
ruff config

ocefpaf committed:
run pre-commit

ocefpaf committed:
data moved

ocefpaf committed:
pre-commit fixes

benjwadams committed:
Refactor formula_terms check, ensure blank formula_terms doesn't pass

benjwadams committed:
Add error if computed_standard_name present, but formula_terms isn't

benjwadams committed:
Merge pull request #1021 from benjwadams/formula_terms_revamp

Formula terms revamp

benjwadams committed:
Merge branch 'develop' into build_api_docs_v2

benjwadams committed:
Merge branch 'develop' into ruff_take_2

benjwadams committed:
Merge pull request #1020 from ocefpaf/ruff_take_2

Ruff take 2

benjwadams committed:
Merge pull request #1008 from ocefpaf/build_api_docs_v2

Build api docs v2

dependabot[bot] committed:
Bump rossjrw/pr-preview-action from 1.3.1 to 1.4.2

Bumps [rossjrw/pr-preview-action](https://github.com/rossjrw/pr-preview-action) from 1.3.1 to 1.4.2.
- [Release notes](https://github.com/rossjrw/pr-preview-action/releases)
- [Commits](https://github.com/rossjrw/pr-preview-action/compare/v1.3.1...v1.4.2)

---
updated-dependencies:
- dependency-name: rossjrw/pr-preview-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

benjwadams committed:
Refactor variable type check to allow for inheritance

benjwadams committed:
Add unsigned integer types added to valid variable types in CF 1.9+

benjwadams committed:
Merge pull request #1022 from ioos/dependabot/github_actions/rossjrw/pr-preview-action-1.4.2

Bump rossjrw/pr-preview-action from 1.3.1 to 1.4.2

benjwadams committed:
Merge pull request #1023 from benjwadams/cf_1_9_add_ubytes_variables

Add unsigned int variables to valid CF 1.9 variable types.

ocefpaf committed:
some pre-commit fixes

ocefpaf committed:
enable bugbear

ocefpaf committed:
fix all but 1 bugbear issues

ocefpaf committed:
skip B019

ocefpaf committed:
skip false positives. See https://github.com/PyCQA/flake8-bugbear/issues/329\#issuecomment-1383938696

ocefpaf committed:
use stacklevel=2

ocefpaf committed:
fix broken link

ocefpaf committed:
let's remove this one for now

benjwadams committed:
Merge pull request #1027 from ocefpaf/fix_broken_link

Fix a broken link and drop a removed one.

benjwadams committed:
Merge pull request #1026 from ocefpaf/pre-commit_fixes

Enable bugbear...

benjwadams committed:
Merge pull request #1025 from ioos/develop

Merge in develop branch in preparation for Compliance Checker v5.1.0

ocefpaf committed:
minor fixes, left over from #1026

ocefpaf committed:
fix more broken links

ocefpaf committed:
remove MemoizedDataset

benjwadams committed:
Merge pull request #1028 from ocefpaf/minor_fix

minor fixes, left over from #1026

ocefpaf committed:
workaround latest urllib3

benjwadams committed:
Merge pull request #1031 from ocefpaf/pin_urllib3

workaround latest urllib3

dependabot[bot] committed:
Bump mamba-org/provision-with-micromamba from 15 to 16

Bumps [mamba-org/provision-with-micromamba](https://github.com/mamba-org/provision-with-micromamba) from 15 to 16.
- [Release notes](https://github.com/mamba-org/provision-with-micromamba/releases)
- [Commits](https://github.com/mamba-org/provision-with-micromamba/compare/v15...v16)

---
updated-dependencies:
- dependency-name: mamba-org/provision-with-micromamba
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

ocefpaf committed:
Merge pull request #1030 from ioos/dependabot/github_actions/mamba-org/provision-with-micromamba-16

Bump mamba-org/provision-with-micromamba from 15 to 16

ocefpaf committed:
redundant

ocefpaf committed:
provision-with-micromamba is deprecated

benjwadams committed:
Pin isodate >= 0.6.1 in requirements.txt

Some builds were failing on isodate requirements due to newer setuptools
not having 2to3 support.  Newer builds of isodate shouldn't have this
issue.

ocefpaf committed:
Merge pull request #1034 from benjwadams/pin_isodate

Pin isodate >= 0.6.1 in requirements.txt

benjwadams committed:
Merge pull request #1033 from ocefpaf/update_umamba_GHA

Update micromamba GitHub Actions

ocefpaf committed:
add top 3 committers as maintainers

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/charliermarsh/ruff-pre-commit: v0.0.267 → v0.0.270](https://github.com/charliermarsh/ruff-pre-commit/compare/v0.0.267...v0.0.270)

ocefpaf committed:
Merge pull request #1035 from ioos/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

ocefpaf committed:
restrict to the upstream feat ver

ocefpaf committed:
update min python

benjwadams committed:
Merge pull request #1029 from ocefpaf/remove_MemoizedDataset

remove MemoizedDataset

ocefpaf committed:
update pre-commits

ocefpaf committed:
remove dups in sets

ocefpaf committed:
add a skip for dups we want to keep

ocefpaf committed:
Merge pull request #1037 from ocefpaf/update_precommit_and_fix

Update precommit and fix

ocefpaf committed:
update

ocefpaf committed:
fix .format calls

ocefpaf committed:
use a server that won't kick us out

ocefpaf committed:
remove all but 1 pkg_resources imports

ocefpaf committed:
remove distutils

ocefpaf committed:
let's skip this one for now

ocefpaf committed:
Merge pull request #1042 from ocefpaf/fix_docs

let's skip this one for now

ocefpaf committed:
update pre-commits

ocefpaf committed:
run ruff fix

ocefpaf committed:
fix E721

ocefpaf committed:
Merge pull request #1038 from ocefpaf/fix_pre-commits

Fix pre commits

dependabot[bot] committed:
Bump actions/checkout from 3 to 4

Bumps [actions/checkout](https://github.com/actions/checkout) from 3 to 4.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

ocefpaf committed:
Merge pull request #1040 from ioos/dependabot/github_actions/actions/checkout-4

Bump actions/checkout from 3 to 4

dependabot[bot] committed:
Bump codecov/codecov-action from 3 to 4

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from 3 to 4.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/v3...v4)

---
updated-dependencies:
- dependency-name: codecov/codecov-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

ocefpaf committed:
Merge pull request #1043 from ioos/dependabot/github_actions/codecov/codecov-action-4

Bump codecov/codecov-action from 3 to 4

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/psf/black: 23.7.0 → 23.9.1](https://github.com/psf/black/compare/23.7.0...23.9.1)
- https://github.com/charliermarsh/ruff-pre-commit → https://github.com/astral-sh/ruff-pre-commit
- [github.com/astral-sh/ruff-pre-commit: v0.0.287 → v0.0.292](https://github.com/astral-sh/ruff-pre-commit/compare/v0.0.287...v0.0.292)
- [github.com/tox-dev/pyproject-fmt: 1.1.0 → 1.2.0](https://github.com/tox-dev/pyproject-fmt/compare/1.1.0...1.2.0)
- [github.com/codespell-project/codespell: v2.2.5 → v2.2.6](https://github.com/codespell-project/codespell/compare/v2.2.5...v2.2.6)

ocefpaf committed:
fix typos

ocefpaf committed:
Merge pull request #1045 from ocefpaf/typos

fix typos

ocefpaf committed:
v4 is beta and was probably mistankely release as stable

ocefpaf committed:
Merge pull request #1046 from ocefpaf/codecov

v4 is beta and was probably mistakenly release as stable

ocefpaf committed:
Merge pull request #1044 from ioos/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

ocefpaf committed:
lint

ocefpaf committed:
add packaging

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/pre-commit/pre-commit-hooks: v4.4.0 → v4.5.0](https://github.com/pre-commit/pre-commit-hooks/compare/v4.4.0...v4.5.0)
- [github.com/psf/black: 23.9.1 → 23.10.1](https://github.com/psf/black/compare/23.9.1...23.10.1)
- [github.com/astral-sh/ruff-pre-commit: v0.0.292 → v0.1.4](https://github.com/astral-sh/ruff-pre-commit/compare/v0.0.292...v0.1.4)
- [github.com/tox-dev/pyproject-fmt: 1.2.0 → 1.4.1](https://github.com/tox-dev/pyproject-fmt/compare/1.2.0...1.4.1)

ocefpaf committed:
Merge pull request #1047 from ioos/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/psf/black: 23.10.1 → 23.11.0](https://github.com/psf/black/compare/23.10.1...23.11.0)
- [github.com/astral-sh/ruff-pre-commit: v0.1.4 → v0.1.6](https://github.com/astral-sh/ruff-pre-commit/compare/v0.1.4...v0.1.6)
- [github.com/tox-dev/pyproject-fmt: 1.4.1 → 1.5.3](https://github.com/tox-dev/pyproject-fmt/compare/1.4.1...1.5.3)

ocefpaf committed:
Merge pull request #1048 from ioos/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

dependabot[bot] committed:
Bump actions/setup-python from 4 to 5

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 4 to 5.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

ocefpaf committed:
Merge pull request #1049 from ioos/dependabot/github_actions/actions/setup-python-5

Bump actions/setup-python from 4 to 5

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/psf/black: 23.11.0 → 23.12.1](https://github.com/psf/black/compare/23.11.0...23.12.1)
- [github.com/astral-sh/ruff-pre-commit: v0.1.6 → v0.1.9](https://github.com/astral-sh/ruff-pre-commit/compare/v0.1.6...v0.1.9)

ocefpaf committed:
Merge pull request #1050 from ioos/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

dependabot[bot] committed:
Bump codecov/codecov-action from 3 to 4

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from 3 to 4.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/v3...v4)

---
updated-dependencies:
- dependency-name: codecov/codecov-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

ocefpaf committed:
Merge pull request #1051 from ioos/dependabot/github_actions/codecov/codecov-action-4

Bump codecov/codecov-action from 3 to 4

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/psf/black: 23.12.1 → 24.1.1](https://github.com/psf/black/compare/23.12.1...24.1.1)
- [github.com/astral-sh/ruff-pre-commit: v0.1.9 → v0.2.0](https://github.com/astral-sh/ruff-pre-commit/compare/v0.1.9...v0.2.0)
- [github.com/tox-dev/pyproject-fmt: 1.5.3 → 1.7.0](https://github.com/tox-dev/pyproject-fmt/compare/1.5.3...1.7.0)

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ocefpaf committed:
Merge pull request #1052 from ioos/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

Karinon committed:
fixed wrong score for two cf-checks which resulted in
appearance of warnings even if there is nothing to warn

Karinon committed:
removed a very verbose debug print statement which resulted in a flooded
CLI

jr3cermak committed:
pytest fixes

 * base.py: Sometimes cfutil is None
 * protocols/netcdf.py:
   - Return False right away if content_type is None
   - Support extended type strings: "application/x-netcdf;ver=4"
 * suite.py: If text for xml fails, look for netcdf one more time
 * Convert suite.run() calls to suite.run_all() calls
 * Update hyrax url
 * Update thredds url
 * Add some development notes for troubleshooting tests with pytest
   using the vcr feature.

jr3cermak committed:
Linter updates

dependabot[bot] committed:
Bump peaceiris/actions-gh-pages from 3 to 4

Bumps [peaceiris/actions-gh-pages](https://github.com/peaceiris/actions-gh-pages) from 3 to 4.
- [Release notes](https://github.com/peaceiris/actions-gh-pages/releases)
- [Changelog](https://github.com/peaceiris/actions-gh-pages/blob/main/CHANGELOG.md)
- [Commits](https://github.com/peaceiris/actions-gh-pages/compare/v3...v4)

---
updated-dependencies:
- dependency-name: peaceiris/actions-gh-pages
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

ocefpaf committed:
Merge pull request #1058 from ioos/dependabot/github_actions/peaceiris/actions-gh-pages-4

Bump peaceiris/actions-gh-pages from 3 to 4

ocefpaf committed:
update pre-commits

ocefpaf committed:
fix deprecation warnings

ocefpaf committed:
fix some lints

ocefpaf committed:
Merge pull request #1039 from ocefpaf/remove_pkg_resources

Remove pkg resources

ocefpaf committed:
add py312

ocefpaf committed:
drop 3.7 and add 3.12 in the tests

ocefpaf committed:
no need to use minor version here

ocefpaf committed:
revert to 3.11 b/c cf_units is broken on 3.12

ocefpaf committed:
missing }

ocefpaf committed:
py38 compat

ocefpaf committed:
Merge pull request #1024 from ocefpaf/pyproject_toml

move to pyproject.toml

ChrisBarker-NOAA committed:
removed last of pkg_resources.

ChrisBarker-NOAA committed:
using importlib_metadata for 3.10_ only

benjwadams committed:
Merge pull request #1061 from ioos/develop

Merge in develop changes for 5.1.1 release preparation

ocefpaf committed:
update pre-commits

ocefpaf committed:
only essential pkg_resource fixes

ocefpaf committed:
sort

ocefpaf committed:
Merge pull request #1060 from ChrisBarker-NOAA/rm_pkg_res

removed last of pkg_resources.

benjwadams committed:
Merge pull request #1056 from uw-farlab/pytest_cleanup

Some pytest fixes

benjwadams committed:
Merge pull request #1053 from Karinon/develop

Score fix in cf 1.6

benjwadams committed:
Fix valid range test after changes in #1053

benjwadams committed:
Fix initial importlib check suite load

Fixes loading initial available checkers via importlib. An invalid
argument had been passed with kwarg `groups` instead of `group`.

benjwadams committed:
Fix valid range with valid min/max check logic
Fixes a logic error in valid_range with valid_min/valid_max which
resulted in erroneous results.

benjwadams committed:
Rename function check_valid_range_or_valid_min_max_present

Renames check function `check_valid_range_or_valid_min_max_present` to
`check_valid_range_and_valid_min_max_present` for clarity.  Existing compliance
checker plugins checked do not reference this function so this appears
to be a safe operation.

benjwadams committed:
Black formatting fixes

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

benjwadams committed:
No-op MockNetCDF __dealloc__ method

Overrides `__dealloc__` method in `MockNetCDF` class to prevent suppress
exception messages on tests via no-op/`pass` operation.

benjwadams committed:
Suppress MockNetCDF __dealloc__ exceptions rather than overriding

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

benjwadams committed:
Merge pull request #1062 from ioos/develop

Final merge prior to 5.1.1 release

benjwadams committed:
Merge pull request #1063 from ioos/master

Merge in current master

ocefpaf committed:
remove lru cache

ocefpaf committed:
update pre-commits

ocefpaf committed:
lint

ocefpaf committed:
don't use a netCDF object as index for caching

ocefpaf committed:
update pre-commits

ocefpaf committed:
fix ioosngdac link

ocefpaf committed:
fix pacioos link

benjwadams committed:
Merge pull request #1069 from ocefpaf/sys_excepthook

Should fix more but now all "Error in sys.excepthook:"

benjwadams committed:
Merge pull request #1070 from ocefpaf/fix_glider_link

Fix some broken links

benjwadams committed:
Merge branch 'main' into Improve-CF-Conformance-5.6-Grid-Mappings-and-Projections

benjwadams committed:
Fixup unused parameter name in cfutil get_grid_mapping_variables

benjwadams committed:
FIXUP: More undef fixes

benjwadams committed:
FIXUP: Add missing regex module

leilabbb committed:
Update cf_1_7.py

reviewed and updated a few lines to use hasatt() and replace var.ncattrs()
